Formula/ibex.rb

```diff
@@ -12,37 +24,82 @@ class Ibex < Formula
     sha256 "41327b6a0da9b8b2ed888c3c2c33d5f6ff061ea13eceb4b9e8b6e660049d624d" => :mountain_lion
   end
 
-  option "with-java", "Build Java bindings for Choco solver."
+  option "with-java", "Enable Java bindings for CHOCO solver."
+  option "with-affine", "Use Affine Arithmetic plugin"
+  option "with-ensta-robotics", "Use Contractors for robotics (SLAM) plugin"
+  option "with-ampl", "Use AMPL file loader plugin"
+  option "with-optim", "Use IbexOpt (Global optimization and System solving) plugin"
+  option "with-param-estim", "Use Parameter Estimation (enhanced Q-intersection algorithm) plugin"
+
+  option "with-debug", "Build Ibex with debug support and disable optimisations."
 
   depends_on :java => ["1.8+", :optional]
   depends_on "bison" => :build
   depends_on "flex" => :build
-  depends_on "pkg-config" => :build
+  depends_on "pkg-config"
+
+  if build.with?("java") && build.with?("ampl")
+    odie "Cannot set options --with-java and --with-ampl simultaneously for now."
+  end
 
   def install
     args = ["--prefix=#{prefix}"]
+    args << "--enable-shared"
+
+    args << "--with-debug" if build.with? "debug"
 
     if build.with? "java"
       args << "--with-jni"
-    else
-      args << "--enable-shared"
     end
 
+    if build.with? "affine"
+      args << "--with-affine"
+      opoo "LinearRelaxAffine class (from option --with-affine) will not be generated " \
```

Comment:
We'd heavily prefer not to throw warnings during compile. Perhaps just make it so if you build with one option the other is invoked as well.